base/compiler/abstractinterpretation.jl

```diff
@@ -1149,15 +1150,29 @@ function abstract_invoke(interp::AbstractInterpreter, argtypes::Vector{Any}, sv:
     nargtype = Tuple{ft, nargtype.parameters...}
     argtype = Tuple{ft, argtype.parameters...}
     result = findsup(types, method_table(interp))
-    if result === nothing
-        return CallMeta(Any, false)
-    end
+    result === nothing && return CallMeta(Any, false)
     method, valid_worlds = result
     update_valid_age!(sv, valid_worlds)
     (ti, env::SimpleVector) = ccall(:jl_type_intersection_with_env, Any, (Any, Any), nargtype, method.sig)::SimpleVector
-    rt, edge = typeinf_edge(interp, method, ti, env, sv)
+    (; rt, edge) = result = abstract_call_method(interp, method, ti, env, false, sv)
     edge !== nothing && add_backedge!(edge::MethodInstance, sv)
-    return CallMeta(rt, InvokeCallInfo(MethodMatch(ti, env, method, argtype <: method.sig)))
+    match = MethodMatch(ti, env, method, argtype <: method.sig)
+    # try constant propagation with early take-in of the heuristics -- since constuctions below seem to be a bit costly
```

Comment:
I don't know if this manual inlining is worth it compared to the cost of duplication. Inference is already very expensive and the `invoke` path is not particularly hot.

Reply:
Ah, I guess you did benchmark this. LGTM then.

Reply:
Yeah, it actually reduced memory allocation and compile time:
```julia
# without manual inlining
julia> @time using Plots
  3.810949 seconds (7.33 M allocations: 513.128 MiB, 6.46% gc time)

# with manual inlining
julia> @time using Plots
  4.252111 seconds (7.01 M allocations: 489.437 MiB, 6.27% gc time)
```

That suggests, some logic within `abstract_call_method_with_const_args` is costly and we may want to refactor it so to gain the same improvements for `abstract_call_gf_by_type` too, but that would be a separate PR.